fv3config/_asset_list.py

```diff
@@ -102,71 +116,45 @@ def get_asset_dict(source_location, source_name, target_location='',
-    """Return an asset_list corresponding to all files within path"""
-    if is_gsbucket_url(path):
-        if gsutil_is_installed():
-            return asset_list_from_gs_bucket(path, target_location=target_location)
-        else:
-            raise DependencyError(f'Optional dependency gsutil not found. Files in {path} will not be copied to {target_location}.')
-    else:
-        return asset_list_from_local_dir(path,
-                                         target_location=target_location,
-                                         copy_method=copy_method)
-
-
-def asset_list_from_local_dir(source_directory, target_location='', copy_method='copy'):
-    """Return asset_list from all files in the local path source_directory. Will
-    recurse to subdirectories within source_directory.
-
-    Args:
-        source_directory (str): path to local directory from which to generate
-            asset_list.
-        target_location (str, optional): target_location used for generated assets.
-            Defaults to '' which is root of run-directory.
-        copy_method (str, optional): copy_method used for generated assets. Defaults
-            to 'copy'.
-
-    Returns:
-        list: a list of asset dictionaries
-    """
-    asset_list = []
-    for root, dirs, files in os.walk(source_directory):
-        if root == source_directory:
-            file_target_location = target_location
-        else:
-            file_target_location = os.path.join(target_location,
-                                                os.path.basename(root))
-        for file in files:
-            asset_list.append(get_asset_dict(os.path.join(source_directory, root),
-                                             file,
-                                             target_location=file_target_location,
-                                             copy_method=copy_method))
-    return asset_list
+def _get_gcloud_fs():
+    return gcsfs.GCSFileSystem(project=_get_gcloud_project())
 
 
-def asset_list_from_gs_bucket(url, target_location=''):
-    """Return asset_list from all files that begin with a google cloud storage url.
-    Will not recurse to files that are in "sub-directories" of the url.
+def asset_list_from_path(location, target_location='', copy_method='copy'):
+    """Return asset_list from all files within a given path.
 
     Args:
-        url (str): google cloud storage url from which to generate asset_list.
+        location (str): local path or google cloud storage url.
         target_location (str, optional): target_location used for generated assets.
             Defaults to '' which is root of run-directory.
+        copy_method ('copy' or 'link', optional): whether to copy or link assets,
+            defaults to 'copy'. If location is a google cloud storage url, this option
+            is ignored and files are copied.
 
     Returns:
         list: a list of asset dictionaries
         """
+    if is_gsbucket_url(location):
```

Comment:
I would recommend creating the filesystem object using the approach from my multiple restart PR: https://github.com/VulcanClimateModeling/fv3net/blob/7c224ffc885f15196c64017ea941784e53af5d93/external/vcm/vcm/misc/rundir.py#L81 

This will allow using any type of filesystem supported by `fsspec` rather than just GCS or local. `fsspec.filesystems('file')` creates a local file system, `fsspec.filesystems('gs')` creates a GCS one.

Reply:
`fsspec.filesystems` is really nice example of using the factory design pattern IMO.